lib/tasks/npm-task.js

```diff
@@ -28,7 +36,27 @@ class NpmTask extends Task {
     return RSVP.resolve(execa('npm', args, { preferLocal: false }));
   }
 
-  checkVersion() {
+  yarn(args) {
+    logger.info('yarn: %j', args);
+    return RSVP.resolve(execa('yarn', args, { preferLocal: false }));
```

Comment:
If `execa` throws, the `yarn` method will violate the promise guarentee of no throws. I think it would be best to wrap the call in the promise constructor, and call resolve on the result of `execa` so throws while calling `execa`

something like:

```js
return new Promise(resolve => resolve(execa('yarn', args, { preferLocal: false }));
```

this more closely mimics the async function semantics as well.